src/tribler/core/components/ipv8/eva/tests/test_scheduler.py

```diff
@@ -0,0 +1,190 @@
+
+
+async def test_schedule(scheduler: Scheduler):
+    scheduler.can_be_send_immediately = Mock(return_value=False)
+
+    scheduler.schedule(transfer=MockTransfer())
+
+    assert not scheduler.eva.start_transfer.called
+    assert len(scheduler.scheduled) == 1
+
+
+async def test_start_transfer(scheduler: Scheduler):
+    scheduler.can_be_send_immediately = Mock(return_value=True)
+    transfer = MockTransfer()
+    transfer.start = Mock(wraps=transfer.start)
+
+    scheduler.schedule(transfer)
+
+    assert transfer.start.called
+    assert not scheduler.scheduled
+
+
+def _fill_test_data(scheduler: Scheduler):
+    scheduler.eva.incoming = {'peer1': MockTransfer()}
+    scheduler.eva.outgoing = {'peer2': MockTransfer()}
+
+    transfers = (
+        MockTransfer(peer='peer1', container=scheduler.eva.incoming),
+        MockTransfer(peer='peer1', container=scheduler.eva.outgoing),
+        MockTransfer(peer='peer2', container=scheduler.eva.incoming),
+        MockTransfer(peer='peer2', container=scheduler.eva.outgoing),
+        MockTransfer(peer='peer3', container=scheduler.eva.incoming),
+        MockTransfer(peer='peer3', container=scheduler.eva.outgoing)
+    )
+
+    scheduler.scheduled = dict.fromkeys(transfers)
+
+
+def _transform_to_str(eva, transfers: Iterable[Transfer]) -> Iterable[str]:
+    """Function transforms transfers to strings like 'peer1_out'"""
+    for transfer in transfers:
+        container = "in" if transfer.container == eva.incoming else "out"
+        yield f'{transfer.peer}_{container}'
+
+
+async def test_transfers_that_can_be_send(scheduler: Scheduler):
+    _fill_test_data(scheduler)
+
+    # Regarding the test data, all scheduled transfer should be ready to send
+    # except transfers for `peer1` and `peer2` because they already exists in
+    # incoming and outgoing containers
+    ready_to_send = scheduler._transfers_that_can_be_send()
+    str_representation = list(_transform_to_str(scheduler.eva, ready_to_send))
+    assert str_representation == ['peer1_out', 'peer2_in', 'peer3_in', 'peer3_out']
+
+
+async def test_send_scheduled(scheduler: Scheduler):
+    _fill_test_data(scheduler)
+
+    # Regarding the test data, all scheduled transfer should be ready to send
+    # except transfers for `peer1` and `peer2` because they already exists in
+    # incoming and outgoing containers
+    #
+    # In this test we will limit amount of `max_simultaneous_transfers` by 4
+    # transfers. In this case only two transfers from scheduled should be send.
+
+    scheduler.eva.settings.max_simultaneous_transfers = 4
+    started = scheduler.send_scheduled()
+    str_representation = list(_transform_to_str(scheduler.eva, started))
+    assert str_representation == ['peer1_out', 'peer2_in']
+
+
+async def test_send_scheduled_task():
+    scheduler = Scheduler(eva=Mock(settings=EVASettings(scheduled_send_interval=0.1)))
+    scheduler.send_scheduled = Mock()
+
+    await asyncio.sleep(0.3)
```

Comment:
**nitpick:** looptest can be used here and in other EVA tests that use `asyncio.sleep` to speed up test execution a bit

Reply:
We use [looptime](https://pypi.org/project/looptime/) for the tests. 

Regarding the documentation:

>Nothing is needed to make async tests run with the fake time, it just works